lib/saml.js

```diff
@@ -161,11 +161,16 @@ SAML.prototype.requestToUrl = function (request, operation, callback) {
         }
 
         var base64 = buffer.toString('base64');
-        var target = self.options.entryPoint + '?';
+
+        var hasQuery = function(url) {
+            return !!(new URL(url).search);
+        }
+
+        var target = self.options.entryPoint + (hasQuery(self.options.entryPoint) ? '&' : '?');
 
         if (operation === 'logout') {
             if (self.options.logoutUrl) {
-                target = self.options.logoutUrl + '?';
+                target = self.options.logoutUrl + (hasQuery(self.options.logoutUrl) ? '&' : '?');
```

Comment:
A more robust way to do this is to use the URL builder itself. https://nodejs.org/docs/latest-v10.x/api/url.html#url_class_urlsearchparams